### PR TITLE
DRILL-6955: storage-jdbc tests improvements

### PR DESCRIPTION
**Pull request by vvysotskyi**
- Remove plugins usage for instantiating test databases;
- Replace derby with h2 database.

For details see [DRILL-6955](https://issues.apache.org/jira/browse/DRILL-6955).

### CODE REVIEW

File: contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java [hunk 1]
```diff
@@ -17,24 +17,101 @@
...
+
+      JdbcStoragePlugin jdbcCaseSensitiveStoragePlugin = new JdbcStoragePlugin(jdbcCaseSensitiveStorageConfig,
+          cluster.drillbit().getContext(), mysqlCaseSensitivePluginName);
+      pluginRegistry.addPluginToPersistentStoreIfAbsent(mysqlCaseSensitivePluginName, jdbcCaseSensitiveStorageConfig, jdbcCaseSensitiveStoragePlugin);
+    }
+  }
+
+  @AfterClass
+  public static void stopMysql() {
+    mysqld.stop();
```
Comment (arina-ielchiieva) [inline contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java]: Maybe we should check for null first?

File: contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithH2IT.java [hunk 2]
```diff
@@ -18,48 +18,98 @@
...
   }
 
   @Test
   public void testCrossSourceMultiFragmentJoin() throws Exception {
-    testNoResult("SET `planner.slice_target` = 1");
-    test("select x.person_id, y.salary from derby.drill_derby_test.person x "
-        + "join %s y on x.person_id = y.person_id ", TABLE_NAME);
+    try {
+      client.alterSession(ExecConstants.SLICE_TARGET, 1);
+      client.queryBuilder()
```
Comment (arina-ielchiieva) [inline contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithH2IT.java]: Not critical but there is `queryBuilder()` method that implicitly performs `client.queryBuilder()`, also there is `run` method for those queries that does not need to return any result (check `ClusterTest` class).

Comment (vvysotskyi) [reply contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithH2IT.java]: Thanks for the advice, used `queryBuilder()` and `run()` where possible.

File: contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java (hunk 1 above)
Comment (vvysotskyi) [reply contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java]: Agree, added a check.

**Comment by arina-ielchiieva** [comment]
+1, please squash the commits.

**Pull request closed without merging**